### PR TITLE
[Term Entry] Python statsmodels: ARMA

### LINKED ISSUE #5641: [Term Entry] Python statsmodels: ARMA
### Reason/inspiration (optional)

We would like a new entry on the `ARMA` term under Python. The entry should go in a new file under `docs/content/python/concepts/statsmodels/terms/arma/arma.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept
- An `Example` section that provides an example demonstrating the concept in use

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by Mofizik**
### Description

Add new term entry `docs/content/python/concepts/statsmodels/terms/arma/arma.md`
Add new `.png` file for the entry `media/arma-forecast-example.png`

Motivation: project on Codeacademy Git & GitHub course

### Issue Solved
Closes #5641

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---
+Title: 'ARMA'
+Description: 'A class of statistical models used for analyzing and forecasting time series data.'
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:3]: We should rewrite the description for more relevant definition as follows:

```suggestion
Description: 'Represent stationary time series data, where statistical properties remain consistent over time.'
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+## Installation
+To run ARMA model you need to install ```statsmodels``` and ```pandas``` libraries using pip:
+```bash
+pip install statsmodels pandas
+```
+
+## Syntax
+
+### ARIMA Model
+To implement ARMA in python ARIMA (Autoregressive integrated moving average) model will be used.
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:34]: We should simply this sentence to make it easier to understand for the learner.

```suggestion
To implement ARMA in Python, the **ARIMA(Autoregressive integrated moving average) model** will be used.
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+```python
+model = ARIMA(endog, order=(p, d, q))
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:36]: We `py` when defining Python code blocks:

```suggestion
```py
model = ARIMA(endog, order=(p, d, q))
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+```
+- endog: The time series data (y axis).
+- order: A tuple (p, d, q) where:
+- - p: Order of the autoregressive (AR) component.
+- - d: Differencing order (set to 0 for ARMA).
+- - q: Order of the moving average (MA) component.
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:42]: We can improve the list formatting for better understanding:

```suggestion
- `endog`: The time series data (y-axis).
- `order`: A tuple (p, d, q) where:
  - `p`: Order of the autoregressive (AR) component.
  - `d`: Differencing order (set to 0 for ARMA).
  - `q`: Order of the moving average (MA) component.
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+### Fitting the Model
+Once the ARIMA model is initialized, the ```.fit()``` method trains it on the data.
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:44]: We should use single backticks for an inline code block.

```suggestion
Once the ARIMA model is initialized, the `.fit()` method trains it on the data.
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+```python
+model_fit = model.fit()
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:46]: We use `py` when defining Python code block:

```suggestion
```py
model_fit = model.fit()
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+```
+The result, ```model_fit```, contains:
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:48]: We should use single backticks for an inline code block.

```suggestion
The result, `model_fit`, contains:
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+Parameters of the fitted ARMA model.
+Methods for forecasting and evaluating the model.
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:51]: This should either be a list or part of a complete sentence.

```suggestion
- Parameters of the fitted ARMA model.
- Methods for forecasting and evaluating the model.
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+### Forecasting
+The .forecast() method generates predictions for a specified number of future steps.
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:53]: We should always enclose exact function/method names in backticks to signify the exact usage:

```suggestion
The `.forecast()` method generates predictions for a specified number of future steps.
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+```python
+forecast = model_fit.forecast(steps=n)
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:55]: We use `py` when defining Python code block:

```suggestion
```py
forecast = model_fit.forecast(steps=n)
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+```
+- steps: Number of future time steps to predict.
+
+### Visualizing
+Matplotlib is used to visualize the historical data and the forecast.
+```python
+plt.plot(data.index, data["temperature"], label="Historical Data")
+plt.plot(future_dates, forecast, label="Forecast", color="red")
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:63]: We use `py` when defining Python code block:

```suggestion
```py
plt.plot(data.index, data["temperature"], label="Historical Data")
plt.plot(future_dates, forecast, label="Forecast", color="red")
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+```
+- ```data.index```: Dates for the historical data.
+- ```forecast```: Forecasted values for the next steps days.
+- ```future_dates```: Generated dates for the forecast period.
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:67]: These are all inline code blocks, so we should use single backticks here:

```suggestion
- `data.index`: Dates for the historical data.
- `forecast`: Forecasted values for the next steps days.
- `future_dates`: Generated dates for the forecast period.
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+
+| Function/Parameter | Purpose |
+| --- | --- |
+| ```ARIMA(endog, order=(p, d, q))```	| Initializes the ARIMA (or ARMA when d=0) model. |
+| ```.fit()```| Fits the ARIMA/ARMA model to the data. |
+| ```.forecast(steps=n)``` | Predicts the next n time points. |
+| ```plt.plot()```	| Plots the time series data and forecast for visualization.|
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:74]: These are all inline code blocks, so we should use single backticks here:

```suggestion
| `ARIMA(endog, order=(p, d, q))`	| Initializes the ARIMA (or ARMA when d=0) model. |
| `.fit()`| Fits the ARIMA/ARMA model to the data. |
| `.forecast(steps=n)` | Predicts the next n time points. |
| `plt.plot()`	| Plots the time series data and forecast for visualization.| 
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+| Function/Parameter | Purpose |
+| --- | --- |
+| ```ARIMA(endog, order=(p, d, q))```	| Initializes the ARIMA (or ARMA when d=0) model. |
+| ```.fit()```| Fits the ARIMA/ARMA model to the data. |
+| ```.forecast(steps=n)``` | Predicts the next n time points. |
+| ```plt.plot()```	| Plots the time series data and forecast for visualization.| 
+
+## Example
+In this example, static data will be generated randomly and forecasted using ARMA model:
+```python
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:78]: We use `py` when defining Python code block:

```suggestion
```py
```

File: content/python/concepts/statsmodels/terms/arma/arma.md [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+plt.figure(figsize=(10, 6))
+plt.plot(data.index, data["temperature"], label="Historical Data")
+future_dates = pd.date_range(start=data.index[-1] + pd.Timedelta(days=1), periods=10)
+plt.plot(future_dates, forecast, label="Forecast", color="red")
+plt.title("ARMA Forecast")
+plt.xlabel("Date")
+plt.ylabel("Temperature (°C)")
+plt.legend()
+plt.grid()
+plt.show()
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/terms/arma/arma.md:107]: We should add more inline comments for easier readability:

```suggestion
import pandas as pd
import numpy as np
from statsmodels.tsa.arima.model import ARIMA
import matplotlib.pyplot as plt

# Generate synthetic time series data
np.random.seed(42)  # Set seed for reproducibility
dates = pd.date_range(start="2023-01-01", periods=100)
temperatures = 20 + np.sin(np.linspace(0, 3 * np.pi, 100)) + np.random.normal(0, 0.5, 100)
data = pd.DataFrame({"date": dates, "temperature": temperatures})
data.set_index("date", inplace=True)

# Fit ARMA model (ARIMA with no differencing)
model = ARIMA(data["temperature"], order=(2, 0, 2))
model_fit = model.fit()

# Forecast for the next 10 days
forecast = model_fit.forecast(steps=10)

# Generate future dates for the forecast
future_dates = pd.date_range(start=data.index[-1] + pd.Timedelta(days=1), periods=10)

# Plot historical data and forecast
plt.figure(figsize=(10, 6))
plt.plot(data.index, data["temperature"], label="Historical Data")
plt.plot(future_dates, forecast, label="Forecast", color="red")
plt.title("ARMA Forecast")
plt.xlabel("Date")
plt.ylabel("Temperature (°C)")
plt.legend()
plt.grid()
plt.show()
```

**Review by PragatiVerma18 — CHANGES_REQUESTED** [review]
Hey @Mofizik, thanks for contributing this entry. I have a few suggestions, please go through them and make the suggested changes. Let me know if you have any questions.

**Review by PragatiVerma18 — APPROVED** [review]
Hey @Mofizik, thanks for promptly addressing the suggested changes. This PR is now good for the next review! 🚀

**Review by Sriparno08 — APPROVED** [review]
Looks good, @Mofizik!
